libr/bin/bfile.c

```diff
@@ -847,3 +848,55 @@ R_API bool r_bin_file_close(RBin *bin, int bd) {
 	}
 	return false;
 }
+
+R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
+	char hash[128], *p;
+	RHash *ctx;
+	ut8* buf;
+	ut64 buf_len = 0;
+	int i;
+	RBinFile *bf = bin->cur;
+	RBinObject *o = bf->o;
+	RIODesc *iod = r_io_desc_get (bin->iob.io, bf->fd);
+
+	if (!file && iod) {
+		file = iod->name;
+	}
+
+	buf_len = r_io_desc_size (iod);
+	if (buf_len > limit)
+		buf_len = limit;
+	buf = calloc (1, buf_len);
+	r_io_desc_read (iod, buf, buf_len);
+	ctx = r_hash_new (false, R_HASH_MD5);
+	for (i = 0; i < buf_len; i ++) {
+		(void)r_hash_do_md5 (ctx, &buf[i], 1);
+	}
+	r_hash_do_end (ctx, R_HASH_MD5);
+	p = hash;
+	for (i = 0; i < R_HASH_SIZE_MD5; i++) {
+		sprintf (p, "%02x", ctx->digest[i]);
+		p += 2;
+	}
+	*p = 0;
+	char fmt[256];
+	snprintf (fmt, sizeof (fmt), "md5:%s", hash);
+	o->info->hashes = r_strbuf_new (fmt);
+	r_hash_free (ctx);
+	ctx = r_hash_new (false, R_HASH_SHA1);
```

Comment:
you can set more than one  hash  in  this r_hash_new(), the  loop will  update all the defined hashes and fill the results for all of them at the end.  no  need to iterate twice